voidcrew/code/game/objects/items/devices/walkman/cassette_deck.dm

```diff
@@ -0,0 +1,107 @@
+/obj/item/device/cassette_deck
+	name = "Dual Cassette Deck"
+	desc = "A Dual Cassette Deck, popular for its ability to copy songs from a cassette. A relic of the old times"
+	icon = 'voidcrew/icons/obj/walkman.dmi'
+	icon_state = "walkman"
+	w_class = WEIGHT_CLASS_SMALL
+	//The cassette that is being copied from
+	var/obj/item/device/cassette_tape/send
+	//List of songs the sender has
+	var/list/sender_list
+	//List of names the Sender has
+	var/list/sender_names
+	//The cassette you are copying to
+	var/obj/item/device/cassette_tape/recieve
+	//List of songs the Reciever has
+	var/list/reciever_list
+	//List of song names the Reciever has
+	var/list/reciever_names
+	//Changes between removal and addition mode
+	var/removal = FALSE
+
+/obj/item/device/cassette_deck/AltClick(mob/user)
+	if(recieve || send)
+		eject_tape(user)
+		return
+	else
+		..()
+
+/obj/item/device/cassette_deck/CtrlClick(mob/user)
+	if(removal == TRUE)
+		removal = FALSE
+		to_chat(user,"You click a button and change the Cassette Deck to splicing mode")
+	else
+		removal = TRUE
+		to_chat(user,"You click a button and change the Cassette Deck to removal mode")
+
+/obj/item/device/cassette_deck/attackby(obj/item/cassette, mob/user)
+	if(istype(cassette, /obj/item/device/cassette_tape))
+		if(!send || !recieve)
+			insert_tape(cassette)
+			playsound(src,'sound/weapons/handcuffs.ogg',20,1)
+			to_chat(user,("You insert \the [cassette] into \the [src]"))
+		else
+			to_chat(user,("Remove a tape first!"))
+
+/obj/item/device/cassette_deck/attack_self(mob/user)
+	..()
+	if(!recieve)
+		to_chat(user,("No Cassette to edit please insert a cassette to edit!"))
+		return
+
+	if(!reciever_names.len && removal == TRUE)
+		to_chat(user,"No songs to remove please change cassette")
+		return
+
+	if(!sender_names.len && removal == FALSE)
```

Comment:
You can add these two if statements into the one below, because that way you won't need to check `removal` 3 different times